contracts/modules/Checkpoint/Voting/APLCR/AdvancedPLCRVotingCheckpoint.sol

```diff
@@ -0,0 +1,1004 @@
+        uint256 _checkpointId
+    )
+    internal
+    {
+        //TODO: Charging Usage cost
+        _startTime = _getStartTime(_startTime);
+        _isEmptyBytes32(_name);
+        _isEmptyString(_proposalTitles);
+        _isGreaterThanZero(_commitDuration, _revealDuration);
+        _isValidLength(_noOfChoices.length, _details.length);
+        uint256 ballotId = ballots.length;
+        ballots.push(Ballot(
+            _checkpointId, uint64(_commitDuration), uint64(_revealDuration), uint64(_startTime), uint24(_noOfChoices.length), uint32(0), false, _name
+        ));
+        //Ballot storage currentBallot = ballots[ballotId];
+        for (uint256 i = 0; i < _noOfChoices.length; i++) {
+            _addProposal(_details[i], _noOfChoices[i], ballots[ballotId], i);
+        }
+        emit CumulativeBallotCreated(
+            ballotId, _checkpointId, _name, _startTime, _commitDuration, _revealDuration, _details, _noOfChoices, _proposalTitles, _choices
+        );
+    } 
+
+    /**
+     * @notice Use to create the ballot
+     * @param _name Name of the ballot (Should be unique)
+     * @param _startTime startTime of the ballot
+     * @param _commitDuration Unix time period till the voters commit there vote
+     * @param _revealDuration Unix time period till the voters reveal there vote starts when commit duration ends
+     * @param _proposalTitles Title of proposal (Comma seperated values)
+     * @param _details Off-chain details related to the proposal
+     * @param _choices Choices of proposals (Comma seperated values)
+     * @param _noOfChoices Array of No. of choices (If it is 0 then it means NAY/YAY ballot type is choosen).
+     * @param _checkpointId Valid checkpoint Id
+     * @param _exemptedAddresses List of addresses not allowed to vote
+     */
+    function createCustomCumulativeBallotWithExemption( 
+        bytes32 _name,
+        uint256 _startTime,
+        uint256 _commitDuration,
+        uint256 _revealDuration,
+        string calldata _proposalTitles,
+        bytes32[] calldata _details,
+        string calldata _choices,
+        uint256[] calldata _noOfChoices,
+        uint256 _checkpointId,
+        address[] calldata _exemptedAddresses
+    )
+    external  // Permission will be taken care at createCustomCumulativeBallot function level
+    {
+        createCustomCumulativeBallot(
+            _name,
+            _startTime,
+            _commitDuration,
+            _revealDuration,
+            _proposalTitles,
+            _details,
+            _choices,
+            _noOfChoices,
+            _checkpointId
+        );         
+        _addExemptedAddresses(_exemptedAddresses, ballots.length - 1);
+    } 
+
+    function _addExemptedAddresses(address[] memory _exemptedAddresses, uint256 _ballotId) internal {
+        for (uint256 i = 0; i < _exemptedAddresses.length; i++) {
+            Ballot storage ballot = ballots[_ballotId];
+            if (_exemptedAddresses[i] != address(0)) 
```

Comment:
do we need this check?

Reply:
Yeah, No hard need just don't want unnecessary storage change.